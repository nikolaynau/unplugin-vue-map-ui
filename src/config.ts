export const libraryName = 'vue-map-ui';

export const allComponents = [
  'VMapAttributionControl',
  'VMapControlPosition',
  'VMapLayersControl',
  'VMapScaleControl',
  'VMapZoomControl',
  'VMapDefaultIcon',
  'VMapDivIcon',
  'VMapDivMarker',
  'VMapIcon',
  'VMapIconMarker',
  'VMapMarker',
  'VMapPinIcon',
  'VMapPinMarker',
  'VMapTileLayer',
  'VMap',
  'VMapPane',
  'VMapArcGisAeroTileLayer',
  'VMapArcGisTileLayer',
  'VMapGoogleTileLayer',
  'VMapMapboxTileLayer',
  'VMapOpenTransportTileLayer',
  'VMapOsmTileLayer'
];
