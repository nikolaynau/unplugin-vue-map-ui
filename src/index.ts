export * from './core/resolvers';
export * from './core/presets';
