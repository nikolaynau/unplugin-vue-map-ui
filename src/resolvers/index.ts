export * from './vue3-map-ui';
