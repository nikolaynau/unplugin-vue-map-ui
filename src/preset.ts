import type { TypeImport } from 'unplugin-vue-components';
import { allComponents, libraryName } from './config';

export const VueMapUiPreset: TypeImport = {
  from: libraryName,
  names: allComponents
};
