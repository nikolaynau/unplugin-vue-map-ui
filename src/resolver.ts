import type { ComponentResolver } from 'unplugin-vue-components';
import { libraryName } from './config';

export function VueMapUiResolver(): ComponentResolver {
  return {
    type: 'component',
    resolve: name => {
      if (name.match(/^VMap[A-Z]?/)) {
        return {
          name,
          from: libraryName
        };
      }
    }
  };
}
