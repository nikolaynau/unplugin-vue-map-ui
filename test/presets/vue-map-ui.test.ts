import { afterEach, describe, expect, it } from 'vitest';
import { build } from 'vite';
import path from 'path';
import { fileURLToPath } from 'url';
import { readFile } from 'fs/promises';
import vue from '@vitejs/plugin-vue';
import { deleteSync } from 'del';
import Components from 'unplugin-vue-components/vite';
import { VueMapUiPreset } from '../../src';

const __dirname = fileURLToPath(new URL('.', import.meta.url));
const appDir = path.resolve(__dirname, '../fixtures/app');
const distDir = path.resolve(appDir, './dist');
const dtsFile = path.resolve(appDir, 'components.d.ts');

describe('VueMapUiPreset', () => {
  afterEach(() => {
    deleteSync(distDir);
    deleteSync(dtsFile);
  });

  it('should be added types in dts', async () => {
    await build({
      root: appDir,
      plugins: [
        vue(),
        Components({
          types: [VueMapUiPreset]
        })
      ],
      build: {
        minify: false,
        rollupOptions: {
          external: ['vue', 'vue-map-ui']
        }
      }
    });

    const data = await readFile(dtsFile, 'utf-8');
    expect(data).toMatchSnapshot();
  });
});
