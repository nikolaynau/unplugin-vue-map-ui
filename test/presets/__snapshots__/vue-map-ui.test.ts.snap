// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueMapUiPreset > should be added types in dts 1`] = `
"/* eslint-disable */
/* prettier-ignore */
// @ts-nocheck
// Generated by unplugin-vue-components
// Read more: https://github.com/vuejs/core/pull/3399
import '@vue/runtime-core'

export {}

declare module '@vue/runtime-core' {
  export interface GlobalComponents {
    VMap: typeof import('vue-map-ui')['VMap']
    VMapArcGisAeroTileLayer: typeof import('vue-map-ui')['VMapArcGisAeroTileLayer']
    VMapArcGisTileLayer: typeof import('vue-map-ui')['VMapArcGisTileLayer']
    VMapAttributionControl: typeof import('vue-map-ui')['VMapAttributionControl']
    VMapGoogleTileLayer: typeof import('vue-map-ui')['VMapGoogleTileLayer']
    VMapLayersControl: typeof import('vue-map-ui')['VMapLayersControl']
    VMapMapboxTileLayer: typeof import('vue-map-ui')['VMapMapboxTileLayer']
    VMapOcmTileLayer: typeof import('vue-map-ui')['VMapOcmTileLayer']
    VMapOsmTileLayer: typeof import('vue-map-ui')['VMapOsmTileLayer']
    VMapScaleControl: typeof import('vue-map-ui')['VMapScaleControl']
    VMapTileLayer: typeof import('vue-map-ui')['VMapTileLayer']
    VMapZoomControl: typeof import('vue-map-ui')['VMapZoomControl']
  }
}
"
`;
